src/main/java/org/folio/rest/impl/base/AbstractEmail.java

```diff
@@ -202,26 +200,54 @@ protected Future<Configurations> lookupConfiguration(Map<String, String> request
           return succeededFuture(config);
         }
         String errorMessage = String.format(ERROR_LOOKING_UP_MOD_CONFIG,
-          url, response.statusCode(), response.bodyAsString());
+          path, response.statusCode(), response.bodyAsString());
         logger.error(errorMessage);
         return failedFuture(new ConfigurationException(errorMessage));
       });
   }
 
-  protected Future<Configurations> validateConfiguration(Configurations configurations) {
-    if (isIncorrectSmtpServerConfig(configurations)) {
-      String errorMessage = String.format(ERROR_MIN_REQUIREMENT_MOD_CONFIG,
-        REQUIREMENTS_CONFIG_SET);
-      logger.error(errorMessage);
-      return failedFuture(new SmtpConfigurationException(errorMessage));
-    }
+  private Future<SmtpConfiguration> copyConfigurationAndDeleteFromModConfig(
+    Configurations configurations, Map<String, String> requestHeaders) {
+
+    return succeededFuture(configurations)
+      .map(EmailUtils::convertSmtpConfiguration)
+      .compose(EmailUtils::validateSmtpConfiguration)
+      .compose(smtpConfigurationService::createSmtpConfiguration)
+      .compose(smtpConfig -> deleteEntriesFromModConfig(smtpConfig, configurations, requestHeaders));
```

Comment:
Looks like we pass `smtpConfig` into `deleteEntriesFromModConfig()` just to return it as a result without any real usage and modification. We also don't really need to wait until all the configurations are deleted, and I don't think we should fail the whole process if we fail to delete one or more configurations. So how about this:
```suggestion
      .onSuccess(smtpConfig -> deleteEntriesFromModConfig(configurations, requestHeaders));
```

`deleteEntriesFromModConfig` won't need to pack all the individual Futures into a CompositeFuture, and won't even need to return anything.